### ISSUE TITLE
November 2019 Planning

### ISSUE THREAD

**Issue by designertyler** [carbon-design-system/carbon#4138]
# Timeline

**Week 1: October 7th, 2019 - October 11th, 2019**
- Project kickoff
- Start release cycle

**Week 2: October 14th, 2019 - October 18th, 2019**

- Project work - no endgame tasks

**Week 3: October 21st, 2019 - October 25th, 2019**

- Project work - no endgame tasks

**Week 4: October 28th, 2019 - November 1st, 2019**
- Patch release
- Pre-endgame planning
- Code freeze by Friday, November 1st

**Week 5: November 4th, 2019 - November 8th, 2019**
- Endgame - All week
- Release Candidate cuts and smoke tests

**Week 6: November 11th, 2019 - November 15th, 2019**
- Release, Endgame done, demo, retro, planning for next release
- Release on Tuesday, November 12th

# Projects

Project assignment goals:
* No single-discipline project teams
* Using RACI (responsible, accountable, consulted, informed) terminology
  * Every project has one Carbon team member who is accountable for the project's completion
  * Carbon team members only responsible for one project at a time
* Prevent project carry-over by getting projects to done done
* As a team, find a way to not neglect support responsibilities

## Design system support (including component enhancements), Website support

* Accountable: @aagonzales 
* Responsible: @jnm2377 , @asudoh 
* Consulted: @alisonjoseph , @jeanservaas , @mattrosno , @shinytoyrobots  (for support definition)
* Informed: 

**To do**: better define support model/project.

<details>
<summary>Outcomes</summary>

* Continuous improvement (process, tooling) to best engage inbound community input
* Not let valuable feedback slip through the cracks  
</details>

<details>
<summary>Scope</summary>

* Bug fixes in the codebase
* Issue triage and backlog grooming three-in-a-box (design, dev, OM)
* Self-serve documentation and FAQ (support audit findings)
* Component enhancement backlog - define scope

*Ready for dev:*

- Low contrast notifications: (https://github.com/carbon-design-system/carbon/issues/4282)
- File uploader bugs (https://github.com/carbon-design-system/carbon/issues/3587)
- 40px search field (https://github.com/carbon-design-system/carbon/issues/3457)
- Tooltip bugs: (https://github.com/carbon-design-system/carbon/issues/3208)
- Copy button bugs (https://github.com/carbon-design-system/carbon/issues/3208)
- Data table skeleton state (https://github.com/carbon-design-system/carbon/issues/3032)
- Inline loading style updates (https://github.com/carbon-design-system/carbon/issues/3040)
- New tab variation (https://github.com/carbon-design-system/carbon/issues/1329)
- React Pagination nav (https://github.com/carbon-design-system/carbon/issues/2634)

*Needs design work first:*

- Modal (https://github.com/carbon-design-system/carbon/issues/2922)
- Search functions: (https://github.com/carbon-design-system/carbon/issues/3513, https://github.com/carbon-design-system/carbon/issues/3489)
- Progress bar (https://github.com/carbon-design-system/carbon/issues/3025)
- Reusable floating menu (https://github.com/carbon-design-system/carbon/issues/1903)
- Phone number field (https://github.com/carbon-design-system/carbon/issues/3948)
- Data table style bugs (https://github.com/carbon-design-system/carbon/issues/3210)
</details>

## Theming (type tokens and size-related constraints) (cont.)

* Accountable: @joshblack 
* Responsible: @laurenmrice , @emyarod  
* Consulted: Jeff Chew, James Dow, Olivia Flory
* Informed: @jeanservaas , @mattrosno 

<details>
<summary>Outcomes</summary>

* Enable expressive typography theming for all Carbon components in a future friendly way to assist Digital Design's use of Carbon
</details>

<details>
<summary>Scope</summary>

* Add support for all tokens in our themes (focus on type tokens, others like spacing could get deferred)
* Add constraints, or update styles, such that updating size-related tokens doesn’t break a component
</details>

## Accessibility (cont.)

* Accountable: @dakahn 
* Responsible: @aledavila , @abbeyhrt 
* Consulted: @aagonzales , Sharon Snider
* Informed: Marc Johlic, Erick Renteria

<details>
<summary>Outcomes</summary>

* 100% component WCAG compliance (if not, documented deviations) to prevent adoption pushback and set adopters up for success
* Document a11y efforts, audits, justifications to demonstrate the large body of work that supports our components and build trust with IBM development teams
* Short term: currently at 32% component coverage towards WCAG compliance, get to 54%
</details>

<details>
<summary>Scope</summary>

* Tackle 12 components and bring them to WCAG compliance
  * Breadcrumb
  * Dropdown
  * Timepicker
  * FormItem
  * Loading
  * TextInput
  * TextArea
  * Tag
  * Slider
  * Select
  * OverflowMenu
  * Notifications
</details>

## Patterns (cont.)

* Accountable: @designertyler 
* Responsible: @janchild , @connor-leech 
* Consulted: @chrisconnors-ibm , @joshblack , @dakahn 
* Informed: Craig Moser, @mattrosno 

<details>
<summary>Outcomes</summary>

* Facilitate alignment around and publish the priority patterns as production-quality design guidance.
</details>

<details>
<summary>Scope</summary>

* Patterns lifecycle and governance
* Production and anatomy guidelines https://github.com/carbon-design-system/carbon-website/issues/488
* Priority patterns consolidation work, based initially on Craig's list:
  * Navigation https://github.com/carbon-design-system/carbon-website/issues/328
  * Forms https://github.com/carbon-design-system/carbon-website/issues/329
  * Notifications https://github.com/carbon-design-system/carbon-website/issues/332
  * Guide [needs issue]
* Audit of existing *main branch* patterns against checklist (Markdown template) to determine work required for completion

Stretch goals:
* Audit of existing *experimental* patterns against checklist (Markdown template) to determine work required for completion
* Audit pattern and create a spreadsheet that tracks the components used in each pattern
</details>

## Component documentation (cont.)

* Accountable: @jeanservaas 
* Responsible: @jillianhowarth , @vpicone , @alisonjoseph (implement template for component page accessibility tab)
* Consulted: @dakahn (accessibility content) , @aagonzales (demo component)
* Informed: Sharon Snider, Marc Johlic, Erick Renteria (all accessibility content)

<details>
<summary>Outcomes</summary>

* Provide relevant developer documentation for our canonical implementation (React) so it's easy/simple to build high-quality web apps with Carbon
* Simplify the experience for designers to fully understand our components (what to use, when to use, how to use) via improved usage guidelines and examples
</details>

<details>
<summary>Scope</summary>

* Provide new accessibility tab on component pages, template for IBMa to populate, facilitate contribution
* Remove vanilla JS component documentation from website
  * Find new home for vanilla JS docs and provide way to find those docs when looking
* Implement demo component
  * Render React components
  * Interactive React source that can be copied/pasted
  * Variant/modifier consolidation
  * Theme switcher
* As a prototype deliverable, improve component page information architecture with supporting designs (e.g. floating anchor links)
* Identify gaps in design guidance
</details>

## Icons (cont.)

* Accountable: @joshblack 
* Responsible: @laurenmrice 
* Consulted: @vpicone 
* Informed: 

**Note:** blocked by the completion of the theming project above.

<details>
<summary>Outcomes</summary>

* Process and workflow to maintain assets (icons, pictograms) isn't manual effort to reduce maintenance overhead of the team going forward
</details>

<details>
<summary>Scope</summary>

* Figure out what is left with pictograms and finish
* General polish/refactoring for internal tooling we rely on that would make it easy to build with icons on the site
* Investigate size issue for projects using TypeScript, could mean multiple entrypoints or figuring out ways to reduce generated icon build
* Define deprecation lifecycle
* Fix bundle and import file size issues (not being able to code split)
</details>

## Theming (colors) (cont.)

* Accountable: @joshblack 
* Responsible: @laurenmrice , @tw15egan 
* Consulted: @aagonzales , @jeanservaas 
* Informed: @mattrosno 

**Note:** blocked by the completion of the icons project above.

<details>
<summary>Outcomes</summary>

* 100% of components theme-able (UI shell) so color token usage is done done
* Form a (documented) opinion on how to apply color to the page (zones? light variants?)
</details>

<details>
<summary>Scope</summary>

* Update components like UI shell to use theming tokens (or not)
* Figure out on design and dev side how we introduce zone concepts (or not)
</details>

**Comment by mattrosno** [comment]
Notes from @designertyler:


## 1. `Website`
#### Squad
- @janchild (lead), @jeanservaas (lead) , @aagonzales, @connor-leech, @laurenmrice, @designertyler, @jillianhowarth, Vince?

#### Primary goals

- Component page work
-- Visual design of on-page navigation component (Jeannie)
-- Layout of component documentation in new one-page format (Jillian)
-- Testing of new component page design (Jillian & Jan)
-- Research Github issue panel inclusion on component pages (Dev)
-- Design for Github issue panel (Lauren)

- Component switcher
-- https://github.com/carbon-design-system/carbon-website/issues/357#issuecomment-538528000
-- Build out code (Vince)
-- Design iterations during build out (Anna)

- Pictograms
-- Design work to finalize (Lauren, Mike)

- Design kit work
-- Incorporating latest Sketch updates (Lauren, Anna) 

- Carbon Comms
-- Design work for release messaging (Jeannie)

- Content strategy and development
-- Axure tool page up on site (Connor)
-- Component template Markdown page (Connor)
-- Review of Support audit findings and plan to enhance existing guidance (Jan, Jillian)
-- Write up roles and responsibilities of the maintainer for Experimental patterns (Connor)

- Theming
-- Next steps TBD 

- Gatsby template feature enhancements
-- Vertical spacing on site to be reviewed by Design (Mike, Sadek, Jeannie, Vince)  
-- Show/hide code snippet for Gatsby template (Vince) 

- Production guidelines (Jeannie)
-- Markdown structure and syntax
-- How to make example images and spec'd images: redlines, enhancing existing guidance for margins, browser images to use, when to use large/small images, how to show details, how to show states, highlighting methods and callouts, how to display stepped content, image templates, asset files (sticker sheet).


#### Stretch goals
- [List stretch goals with links to issues]
-- Dev research approach for on-page navigation component (Alison)
-- Audit of Watson Customer Engagement content to determine what can be brought over to Carbon (JillIan/Jan)
-- Planning for Hybrid Cloud voice & tone guide content moving into Carbon (@janchild & Tom Waterton) 
-- Design systems high-level audits of content covered and content approaches cont'd (@janchild)


#### Deferred goals
- [List deferred goals with links to issues]

## 2. Patterns
#### Squad
- @mattrosno, @mjabbink, Maui, @designertyler, @connor-leech, @janchild 

#### Primary goals

- Patterns lifecycle and governance (@mjabbink, @mattrosno) 
- Priority patterns consolidation work, based initially on Craig's list (Tyler, Connor, Maui, Jan)
- Audit of existing main branch patterns against checklist (Markdown template) to determine work required for completion (Tyler)
- Audit of existing experimental patterns against checklist (Markdown template) to determine work required for completion (Tyler)
- Navigation patterns updated and assessed for production readiness (Tyler) https://github.com/carbon-design-system/carbon-website/issues/328
- Audit pattern and create a spreadsheet that tracks the components used in each pattern. (Tyler)

#### Stretch goals
- [List stretch goals with links to issues]

#### Deferred goals
- [List deferred goals with links to issues]

## 3. Plex
#### Squad
- Vince, Andrea, Abbey

#### Primary goals
- Onboard new devs to the Plex project
- Improve contribution documentation/guides
- Add CI, make updates/new languages easier
- Look for opportunities to refactor build pipeline with the goal of making contributions/updates easier

#### Stretch goals
- Variable fonts improvements
	- Proper sub-setting
	- formal integration with sass styles/library
	- Documentation and guidance around usage, cost, and benefits
- Experimental CDN for design ecosystem/carbon sites hosted on IBM cloud

#### Deferred goals
- Public CDN for Plex


## 4. Component Enhancement
#### Squad
- Anna, Lauren, Akira, Andrew

#### Primary goals
Component Enhancements:

*Ready for dev:*
- Low contrast notifications: (https://github.com/carbon-design-system/carbon/issues/4282)
- File uploader bugs (https://github.com/carbon-design-system/carbon/issues/3587)
- 40px search field (https://github.com/carbon-design-system/carbon/issues/3457)
- Tooltip bugs: (https://github.com/carbon-design-system/carbon/issues/3208)
- Copy button bugs (https://github.com/carbon-design-system/carbon/issues/3208)
- Data table skeleton state (https://github.com/carbon-design-system/carbon/issues/3032)
- Inline loading style updates (https://github.com/carbon-design-system/carbon/issues/3040)
- New tab variation (https://github.com/carbon-design-system/carbon/issues/1329)
- React Pagination nav (https://github.com/carbon-design-system/carbon/issues/2634)

*Needs design work first:*
- Modal (https://github.com/carbon-design-system/carbon/issues/2922)
- Search functions: (https://github.com/carbon-design-system/carbon/issues/3513, https://github.com/carbon-design-system/carbon/issues/3489)
- Progress bar (https://github.com/carbon-design-system/carbon/issues/3025)
- Reusable floating menu (https://github.com/carbon-design-system/carbon/issues/1903)
- Phone number field (https://github.com/carbon-design-system/carbon/issues/3948)
- Data table style bugs (https://github.com/carbon-design-system/carbon/issues/3210)

#### Stretch goals
*Ready for dev:*

*Needs design work first:*
- V10 Skeleton state style (https://github.com/carbon-design-system/carbon/issues/3897)

#### Deferred goals
- [List deferred goals with links to issues]

**Comment by shinytoyrobots** [comment]
Once this is "set", can we also write it up as a wiki page at https://github.com/carbon-design-system/carbon/wiki (at least until we confirm the "single source of truth" destination).

**Comment by mattrosno** [comment]
@shinytoyrobots yes. I'm looking to have this ready for endgame team by end of day tomorrow (Tuesday) so they can proceed with the Medium article.
